storage_service/locations/models/package.py

```diff
@@ -1588,6 +1587,51 @@ def backlog_transfer(self, origin_location, origin_path):
         self.status = Package.UPLOADED
         self.save()
 
+    def index_file_data_from_aip_mets(self):
+        """
+        Attempts to read an Archivematica AIP METS file inside this
+        package, then uses the retrieved metadata to generate one entry in the
+        File table in the database for each file inside the package.
+
+        :raises StorageException: if the transfer METS cannot be found,
+            or if required elements are missing.
+        TODO/QUESTION: what happens if the file has been placed in backlog and
+            indexed via ``index_file_data_from_transfer_mets``? Should we
+            update an existing ``File`` instance if there is one? Otherwise we
+            may have multiple ``File`` instances for a single file.
+        """
+        aip_dir_name = os.path.basename(os.path.splitext(self.full_path)[0])
+        relative_path = os.path.join(aip_dir_name, "data", "METS." + self.uuid + ".xml")
+        path_to_mets, temp_dir = self.extract_file(relative_path)
```

Comment:
Comment from @hwesta: Has this been tested with compressed & uncompressed AIPs? Does it work with AIPs not stored in the local filesystem? Since this is happening asynchronously, what happens when the AIP storage operation completes during this thread? eg self.full_path may not be accurate, or may change partway through